### PR TITLE
Feature 888/retention report service

### PR DESCRIPTION
**Pull request by warnerj-oci**
Create service to generate data for retention reports

### CODE REVIEW

File: server/src/main/java/com/objectcomputing/checkins/services/memberprofile/retentionreport/RetentionReportServicesImpl.java [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+@Singleton
+public class RetentionReportServicesImpl implements RetentionReportServices {
+
+    private final MemberProfileServices memberProfileServices;
+
+    public RetentionReportServicesImpl(MemberProfileServices memberProfileServices) {
+        this.memberProfileServices = memberProfileServices;
+    }
+
+    @Override
```
Comment (bhatiam12) [inline server/src/main/java/com/objectcomputing/checkins/services/memberprofile/retentionreport/RetentionReportServicesImpl.java:21]: do we need any validations here like only admin can run the report ?

Comment (warnerj-oci) [reply server/src/main/java/com/objectcomputing/checkins/services/memberprofile/retentionreport/RetentionReportServicesImpl.java:21]: Yes, we probably do. It wasn't in the AC, so I guess I missed it.

File: server/src/main/java/com/objectcomputing/checkins/services/memberprofile/retentionreport/RetentionReportServicesImpl.java [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+                case WEDNESDAY:
+                    startDate = periodStartDate.plusDays(3);
+                    break;
+                case THURSDAY:
+                    startDate = periodStartDate.plusDays(2);
+                    break;
+                case FRIDAY:
+                    startDate = periodStartDate.plusDays(1);
+                    break;
+                default:
```
Comment (jahanner) [inline server/src/main/java/com/objectcomputing/checkins/services/memberprofile/retentionreport/RetentionReportServicesImpl.java:63]: Do you not need a SATURDAY in here? Or is the periodStartDate Sat by default?

Comment (warnerj-oci) [reply server/src/main/java/com/objectcomputing/checkins/services/memberprofile/retentionreport/RetentionReportServicesImpl.java:63]: Right, the default is Saturday, the other cases are adjusting to the Saturday of the week.

File: server/src/main/java/com/objectcomputing/checkins/services/memberprofile/retentionreport/RetentionReportServicesImpl.java (hunk 1 above)
Comment (warnerj-oci) [reply server/src/main/java/com/objectcomputing/checkins/services/memberprofile/retentionreport/RetentionReportServicesImpl.java:25]: Added validation

**Review by jahanner — APPROVED** [review]

**Review by mkimberlin — APPROVED** [review]
eeeeexcellent... yeah...if this turn out the right numbers (which i think it should), I will do a happy dance.

**Pull request merged**